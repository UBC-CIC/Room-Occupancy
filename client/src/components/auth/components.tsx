import React from "react";
import {
  View,
  Image,
  useTheme,
  useAuthenticator,
  Heading,
} from "@aws-amplify/ui-react";
import { useAppConfig } from "../../providers/ConfigProvider";

type Props = {};

const components = {
  Header() {
    const { config } = useAppConfig();
    const { tokens } = useTheme();

    return (
      <View textAlign="center" padding={tokens.space.large}>
        <Image alt="Logo" src={config.image} />
      </View>
    );
  },
  SignIn: {
    Header() {
      const { tokens } = useTheme();

      return (
        <Heading
          padding={`${tokens.space.xl} 0 0 ${tokens.space.xl}`}
          level={3}
        >
          Sign in to your account
        </Heading>
      );
    },
  },
};

export default components;
