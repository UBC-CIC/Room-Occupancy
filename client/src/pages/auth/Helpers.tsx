import {
  signOut,
  fetchUserAttributes,
  updateUserAttributes,
  type UpdateUserAttributesOutput,
} from "aws-amplify/auth";
import { Navigate } from "react-router-dom";

export async function handleSignOut() {
  try {
    await signOut();
  } catch (error) {
    console.error("error signing out: ", error);
  } finally {
    return <Navigate to="/" />;
  }
}

// https://docs.amplify.aws/gen2/build-a-backend/auth/manage-user-profile/#retrieve-user-attributes
export async function handleFetchUserAttributes() {
  try {
    const userAttributes = await fetchUserAttributes();
    return userAttributes;
  } catch (error) {
    console.log(error);
  }
}

export async function handleUpdateNameAttributes(updatedName: string) {
  try {
    const attributes = await updateUserAttributes({
      userAttributes: {
        name: updatedName,
      },
    });
  } catch (error) {
    console.log(error);
  }
}
