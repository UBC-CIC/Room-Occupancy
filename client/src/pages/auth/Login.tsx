import * as React from "react";
import { Button, Form, Segment, Header, Input } from "semantic-ui-react";
import "../../App.css";
import { Link } from "react-router-dom";

export interface ILoginProps {}

export function Login(props: ILoginProps) {
  return (
    <div className="container">
      <Segment compact padded="very">
        <Header color="blue" as="h1" textAlign="center">
          <Header.Content>Welcome</Header.Content>
          <Header.Subheader color="blue">Login with Email</Header.Subheader>
        </Header>
        <Form>
          <Form.Field>
            <label>Name</label>
            <Input icon="user" iconPosition="left" placeholder="Name" />
          </Form.Field>
          <Form.Field>
            <label>Email</label>
            <Input
              type="email"
              icon="mail"
              iconPosition="left"
              placeholder="Email"
            />
          </Form.Field>
          <Form.Field>
            <label>Password</label>
            <Input
              type="password"
              icon="lock"
              iconPosition="left"
              placeholder="Password"
            />
          </Form.Field>
          <div className="flexContainer">
            <Button color="blue" type="submit">
              Login
            </Button>
          </div>
        </Form>
      </Segment>
    </div>
  );
}
