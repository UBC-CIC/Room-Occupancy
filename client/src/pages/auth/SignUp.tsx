import * as React from "react";
import {
  Button,
  Checkbox,
  Form,
  Segment,
  Header,
  Input,
  Divider,
} from "semantic-ui-react";
import "../../App.css";
import { Link } from "react-router-dom";

export interface ISignUpProps {}

export function SignUp(props: ISignUpProps) {
  return (
    <div className="container">
      <Segment compact padded="very">
        <Header color="blue" as="h1" textAlign="center">
          <Header.Content>Sign Up</Header.Content>
          <Header.Subheader color="blue">
            Let's get started, Sign Up!
          </Header.Subheader>
        </Header>
        <Form>
          <Form.Field>
            <label>Name</label>
            <Input icon="user" iconPosition="left" placeholder="Name" />
          </Form.Field>
          <Form.Field>
            <label>Email</label>
            <Input
              type="email"
              icon="mail"
              iconPosition="left"
              placeholder="Email"
            />
          </Form.Field>
          <Form.Field>
            <label>Password</label>
            <Input
              type="password"
              icon="lock"
              iconPosition="left"
              placeholder="Password"
            />
          </Form.Field>
          <Form.Field>
            <Checkbox label="I agree to the Terms and Conditions" />
          </Form.Field>
          <div className="flexContainer">
            <Button color="blue" type="submit">
              Submit
            </Button>
          </div>
        </Form>
        <Divider horizontal>Or</Divider>
        <Header as="h4" textAlign="center">
          Already have an account?
          {<Link to="/login"> Login</Link>}
        </Header>
      </Segment>
    </div>
  );
}
