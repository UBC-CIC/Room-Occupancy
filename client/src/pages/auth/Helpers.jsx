import { signOut } from "aws-amplify/auth";

export async function handleSignOut() {
  try {
    await signOut();
  } catch (error) {
    console.error("error signing out: ", error);
  }
}
