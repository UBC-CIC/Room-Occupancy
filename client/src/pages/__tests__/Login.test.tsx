import * as React from "react";
import { render } from "@testing-library/react";
import { Login } from "../auth/Login";

describe("<Login />", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Login />);

    expect(asFragment()).toMatchSnapshot();
  });
});
