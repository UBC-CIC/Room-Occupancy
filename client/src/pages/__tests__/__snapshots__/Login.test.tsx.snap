// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="ui compact very padded segment"
    >
      <h1
        class="ui blue center aligned header"
      >
        <div
          class="content"
        >
          Welcome
        </div>
        <div
          class="sub header"
          color="blue"
        >
          Login with Email
        </div>
      </h1>
      <form
        class="ui form"
      >
        <div
          class="field"
        >
          <label>
            Name
          </label>
          <div
            class="ui left icon input"
          >
            <input
              placeholder="Name"
              type="text"
            />
            <i
              aria-hidden="true"
              class="user icon"
            />
          </div>
        </div>
        <div
          class="field"
        >
          <label>
            Email
          </label>
          <div
            class="ui left icon input"
          >
            <input
              placeholder="Email"
              type="email"
            />
            <i
              aria-hidden="true"
              class="mail icon"
            />
          </div>
        </div>
        <div
          class="field"
        >
          <label>
            Password
          </label>
          <div
            class="ui left icon input"
          >
            <input
              placeholder="Password"
              type="password"
            />
            <i
              aria-hidden="true"
              class="lock icon"
            />
          </div>
        </div>
        <div
          class="flexContainer"
        >
          <button
            class="ui blue button"
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
